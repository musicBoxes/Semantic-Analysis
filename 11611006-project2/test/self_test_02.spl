int valid_condition_test(){
	int a = 0;
	char b = '9', bb = 8.0;
	float c = 8.0;
	
	if (a) return 0;
	while (a) break;
	
	// condition is not int
	if (b) return 0;
	while (b) break;
	
	// condition is not int
	if (c) return 0;
	while (c) break;
	return 0;
}