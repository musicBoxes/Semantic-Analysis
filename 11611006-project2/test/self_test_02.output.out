Error type 5 at Line 3: unmatching type on both sides of assignment
Semantic Error at line 10: Use non-int type variable as condition.
Semantic Error at line 11: Use non-int type variable as condition.
Semantic Error at line 14: Use non-int type variable as condition.
Semantic Error at line 15: Use non-int type variable as condition.
